packages/react/src/components/Tree/Tree.tsx

```diff
@@ -0,0 +1,271 @@
+
+export interface TreeSlotClassNames {
+  item: string
+}
+
+export interface TreeProps extends UIComponentProps, ChildrenComponentProps {
+  /** Accessibility behavior if overridden by the user. */
+  accessibility?: Accessibility
+
+  /** Initial activeIndex value. */
+  defaultActiveItemsList?: ShorthandCollection<TreeItemProps>
+
+  /** Only allow one subtree to be open at a time. */
+  exclusive?: boolean
+
+  /** Shorthand array of props for Tree. */
+  items?: ShorthandCollection<TreeItemProps>
+
+  /**
+   * A custom render function for the title slot.
+   *
+   * @param {React.ReactType} Component - The computed component for this slot.
+   * @param {object} props - The computed props for this slot.
+   * @param {ReactNode|ReactNodeArray} children - The computed children for this slot.
+   */
+  renderItemTitle?: ShorthandRenderFunction
+}
+
+export interface TreeState {
+  activeItems: Map<ShorthandValue<TreeItemProps>, { open: boolean; element: HTMLElement }>
+}
+
+class Tree extends UIComponent<WithAsProp<TreeProps>, TreeState> {
+  static create: Function
+
+  static displayName = 'Tree'
+
+  static className = 'ui-tree'
+
+  static slotClassNames: TreeSlotClassNames = {
+    item: `${Tree.className}__item`,
+  }
+
+  static propTypes = {
+    ...commonPropTypes.createCommon({
+      content: false,
+    }),
+    activeItemsList: customPropTypes.every([
+      customPropTypes.disallow(['children']),
+      PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.number), PropTypes.number]),
+    ]),
+    defaultActiveItemsList: customPropTypes.every([
+      customPropTypes.disallow(['children']),
+      PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.number), PropTypes.number]),
+    ]),
+    exclusive: PropTypes.bool,
+    items: customPropTypes.collectionShorthand,
+    renderItemTitle: PropTypes.func,
+    rtlAttributes: PropTypes.func,
+    onactiveItemsListChange: PropTypes.func,
+  }
+
+  static defaultProps = {
+    as: 'div',
+    accessibility: treeBehavior,
+  }
+
+  static autoControlledProps = ['activeItemsList']
+
+  itemRefs = []
+
+  constructor(props, context) {
+    super(props, context)
+
+    const activeItems = new Map()
+
+    if (props.items) {
+      const setItemsLevelAndSize = (items = props.items, level = 1, parent?) => {
+        items.forEach((item: ShorthandValue<TreeItemProps>, index: number) => {
+          item['level'] = level
+          item['siblings'] = items
+          item['index'] = index
+          if (parent) {
+            item['parent'] = parent
+          }
+          if (!item['id']) {
+            item['id'] = `${parent ? parent['id'] : ''}${index}`
+          }
+          if (item['items']) {
+            setItemsLevelAndSize(item['items'], level + 1, item)
+            activeItems.set(item['id'], { open: false })
+          }
+        })
+      }
+      setItemsLevelAndSize()
+    }
+
+    this.state = { activeItems }
```

Comment:
I don't see `getDerivedStateFromProps()` or `componentDidUpdate()`, it means that on any props update it will be broken

Reply:
```tsx
import React from 'react'
import { Button, Divider, Tree } from '@stardust-ui/react'

const TreeExampleShorthand = () => {
  const [items, setItems] = React.useState([
    {
      key: '1',
      title: '1',
    },
  ])

  return (
    <>
      <Button
        content="Add item"
        onClick={() => {
          const newItem = {
            key: items.length + 1,
            title: items.length + 1,
          }

          setItems([...items, newItem])
        }}
      />

      <Divider />
      <Tree items={items} />
    </>
  )
}

export default TreeExampleShorthand
```